File: app/src/main/java/org/checklist/comics/comicschecklist/util/CCLogger.java
package org.checklist.comics.comicschecklist.util;

import android.util.Log;

public class CCLogger {

    private static boolean DEBUG = false;

    public static void v(String tag, String msg) {
        if (DEBUG) {
            Log.v(tag, msg);
        }
    }

    public static void d(String tag, String msg) {
        if (DEBUG) {
            Log.d(tag, msg);
        }
    }

    public static void i(String tag, String msg) {
        if (DEBUG) {
            Log.i(tag, msg);
        }
    }

    public static void w(String tag, String msg) {
        if (DEBUG) {
            Log.w(tag, msg);
        }
    }

    public static void w(String tag, String msg, Exception exception) {
        if (DEBUG) {
            Log.w(tag, msg, exception);
        }
    }

    public static void e(String tag, String msg) {
        if (DEBUG) {
            Log.e(tag, msg);
        }
    }
}

File: app/src/main/java/org/checklist/comics/comicschecklist/log/CCLogger.kt
package org.checklist.comics.comicschecklist.log

import android.util.Log

/**
 * Support class used to log info.
 * TODO migrate to LoggerDelegate
 */
object CCLogger {

    private val DEBUG = false

    fun v(tag: String, msg: String) {
        if (DEBUG) {
            Log.v(tag, msg)
        }
    }

    fun d(tag: String, msg: String) {
        if (DEBUG) {
            Log.d(tag, msg)
        }
    }

    fun i(tag: String, msg: String) {
        if (DEBUG) {
            Log.i(tag, msg)
        }
    }

    fun w(tag: String, msg: String) {
        if (DEBUG) {
            Log.w(tag, msg)
        }
    }

    fun w(tag: String, msg: String, exception: Exception) {
        if (DEBUG) {
            Log.w(tag, msg, exception)
        }
    }

    fun e(tag: String, msg: String) {
        if (DEBUG) {
            Log.e(tag, msg)
        }
    }
}

